pkg/startup/v12/startup.go

```diff
@@ -244,3 +244,9 @@ func (s *startup) writeFiles(role api.AgentPoolProfileRole, w writers.Writer, ho
 
 	return w.Close()
 }
+
+// WriteSearchDomain queries for the search domain and writes to
+// /etc/dhcp/dhclient-eth0.conf.  This is for private api clusters only
+func (s *startup) WriteSearchDomain(ctx context.Context, log *logrus.Entry) error {
+	return fmt.Errorf("not supported in this release")
```

Comment:
might be better to return nil here incase we need to patch a prod cluster with a newer startup image (for an unrelated bug). in this case we would want this to be a no-op, not fail

Reply:
sgtm